From c8748463283738aacf323eb3f7853e6533d941dc Mon Sep 17 00:00:00 2001
From: Stanislav Zaikin <zstaseg@gmail.com>
Date: Thu, 30 Jul 2026 10:54:21 +0200
Subject: [PATCH 08679/15333] arm64: dts: qcom: taoyao: Drop
 adsp_rpc_remote_heap_mem override

taoyao originally redefined adsp_rpc_remote_heap_mem at a fixed address
to avoid overlapping with cdsp_mem, since kodiak.dtsi's version was a
fixed-address at the time.

kodiak.dtsi now defines adsp_rpc_remote_heap_mem as a dynamically
allocated "shared-dma-pool" region instead, which removes the fixed
address entirely and so cannot overlap with cdsp_mem. This makes the
override unnecessary.

Remove the taoyao-local /delete-node/ and static reg-based redefinition,
and inherit the definition from kodiak.dtsi instead.

Fixes: 28705df5d0df ("arm64: dts: qcom: sm7325: Add Xiaomi 12 Lite 5G (taoyao) DTS")

Signed-off-by: Stanislav Zaikin <zstaseg@gmail.com>
Reviewed-by: Konrad Dybcio <konrad.dybcio@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20260730085422.631896-1-zstaseg@gmail.com
Signed-off-by: Bjorn Andersson <andersson@kernel.org>
---
 arch/arm64/boot/dts/qcom/sm7325-xiaomi-taoyao.dts | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sm7325-xiaomi-taoyao.dts b/arch/arm64/boot/dts/qcom/sm7325-xiaomi-taoyao.dts
index 6f8bc65bb99b..87f267aaeec0 100644
--- a/arch/arm64/boot/dts/qcom/sm7325-xiaomi-taoyao.dts
+++ b/arch/arm64/boot/dts/qcom/sm7325-xiaomi-taoyao.dts
@@ -27,7 +27,6 @@
 
 /* The following reserved memory regions have different addresses or sizes */
 /delete-node/ &adsp_mem;
-/delete-node/ &adsp_rpc_remote_heap_mem;
 /delete-node/ &cdsp_mem;
 /delete-node/ &rmtfs_mem;
 
@@ -203,11 +202,6 @@ neuron_block_1_mem: neuron-block@d7f80000 {
 			no-map;
 		};
 
-		adsp_rpc_remote_heap_mem: adsp-rpc-remote-heap@e0400000 {
-			reg = <0x0 0xe0400000 0x0 0x800000>;
-			no-map;
-		};
-
 		framebuffer_reserved: framebuffer@e1000000 {
 			reg = <0x0 0xe1000000 0x0 (1080 * 2400 * 4)>;
 			no-map;
-- 
2.55.0

